plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -0,0 +1,244 @@
+import           Development.IDE.Types.Logger         (Priority (..),
+                                                       cmapWithPrio, logWith,
+                                                       (<+>))
+import           GHC.Generics                         (Generic)
+import           Ide.Plugin.RangeMap                  (RangeMap)
+import qualified Ide.Plugin.RangeMap                  as RangeMap
+import           Ide.PluginUtils                      (getNormalizedFilePath,
+                                                       handleMaybeM,
+                                                       pluginResponse)
+import           Ide.Types                            (PluginDescriptor (..),
+                                                       PluginId (..),
+                                                       PluginMethodHandler,
+                                                       defaultPluginDescriptor,
+                                                       mkPluginHandler)
+import           Language.LSP.Types                   (CodeAction (..),
+                                                       CodeActionKind (CodeActionRefactorRewrite),
+                                                       CodeActionParams (..),
+                                                       Command, List (..),
+                                                       Method (..),
+                                                       SMethod (..),
+                                                       TextEdit (..),
+                                                       WorkspaceEdit (WorkspaceEdit),
+                                                       fromNormalizedUri,
+                                                       normalizedFilePathToUri,
+                                                       type (|?) (InR))
+import qualified Language.LSP.Types.Lens              as L
+data Log
+  = LogShake Shake.Log
+  | LogCollectedRecordSelectors [RecordSelectorExpr]
+  | LogTextEdits [TextEdit]
+
+instance Pretty Log where
+  pretty = \case
+    LogShake shakeLog -> pretty shakeLog
+    LogCollectedRecordSelectors recs -> "Collected record selectors:" <+> pretty recs
+
+data CollectRecordSelectors = CollectRecordSelectors
+                    deriving (Eq, Show, Generic)
+
+instance Hashable CollectRecordSelectors
+instance NFData CollectRecordSelectors
+
+data CollectRecordSelectorsResult = CRSR
+  { recordInfos       :: RangeMap RecordSelectorExpr
+  , enabledExtensions :: [Extension]
+  }
+  deriving (Generic)
+
+instance NFData CollectRecordSelectorsResult
+
+instance Show CollectRecordSelectorsResult where
+  show _ = "<CollectRecordsResult>"
+
+type instance RuleResult CollectRecordSelectors = CollectRecordSelectorsResult
+
+-- | Where we store our collected record selectors
+data RecordSelectorExpr =
+    RecordSelectorExpr { -- | the location of the matched record selector, and record
```

Comment:
generally this haddock reads rather weirdly, like you meant it to flow between the documentation on the various fields? I'm not quite sure what you're trying to do here

Reply:
I'm using the documentation for record fields https://haskell-haddock.readthedocs.io/en/latest/markup.html#constructors-and-record-fields